examples/fluids/src/setupts.c

```diff
@@ -144,8 +157,8 @@ PetscErrorCode IFunction_NS(TS ts, PetscReal t, Vec Q, Vec Q_dot, Vec G, void *u
   PetscCall(DMGetLocalVector(user->dm, &G_loc));
 
   // Update time dependent data
+  PetscCall(UpdateBoundaryValues(user, Q_loc, t));
   if (user->time != t) {
-    PetscCall(DMPlexInsertBoundaryValues(user->dm, PETSC_TRUE, Q_loc, t, NULL, NULL, NULL));
     if (user->phys->solution_time_label) {
       CeedOperatorContextSetDouble(user->op_ifunction, user->phys->solution_time_label, &t);
     }
```

Comment:
Would it make sense to move this logic and below inside `UpdateBoundaryValues` (perhaps with a new name/args)?

Reply:
Maybe, but I think the flexibility here is nice. For example the `CollectStatistics` function doesn't need to check whether the time has updated to it's label. 

I think what could be nice is having a dedicated function for updating the label? Something like:
```c
PetscErrorCode UpdateContextLabel(PetscScalar *previous_value, PetscScalar update_value, CeedOperator op, CeedContextFieldLabel label) {
  PetscFunctionBeginUser;

  if (*previous_value != update_value) {
    if (label) {
      CeedOperatorContextSetDouble(op, label, &update_value);
    }
    *previous_value = update_value;
  }
  PetscFunctionReturn(0);
}
```

Having a `CeedOperatorContextGetDouble` would be useful here, instead of having to manually track and update the label value.

Reply:
Adding that (and friends) should be pretty straightforward and seems useful

Reply:
Made a new issue for getting label data: #1144

Reply:
Added the `UpdateContextLabel` function